frontend/src/routes/Applications/ApplicationDetails/ApplicationTopology/model/applicationSubscription.js

```diff
@@ -1,19 +1,20 @@
 /* Copyright Contributors to the Open Cluster Management project */
 
-import { cloneDeep, get, includes, isEmpty } from 'lodash'
+import { cloneDeep, get, isEmpty } from 'lodash'
 import { listResources } from '../../../../../resources/utils/resource-request'
+import { getSubscriptionAnnotations, isLocalSubscription } from '../../../helpers/subscriptions'
 
 const EVERYTHING_CHANNEL = '__ALL__/__ALL__//__ALL__/__ALL__'
 export const ALL_SUBSCRIPTIONS = '__ALL__/SUBSCRIPTIONS__'
 const NAMESPACE = 'metadata.namespace'
 
 export const getSubscriptionApplication = async (model, app, selectedChannel, recoilStates) => {
     // get subscriptions to channels (pipelines)
-    let subscriptionNames = get(app, 'metadata.annotations["apps.open-cluster-management.io/subscriptions"]')
-    if (subscriptionNames && subscriptionNames.length > 0) {
+    let subscriptionNames = getSubscriptionAnnotations(app)
+    if (subscriptionNames.length > 0) {
         // filter local hub subscription
-        const filteredSubscriptions = subscriptionNames.split(',').filter((subscriptionName) => {
-            return !includes(subscriptionName, '-local')
```

Comment:
This is the main part that was causing the issue - that we were using includes instead of endsWith. Took this opportunity to refactor the code further.